File: server/src/knowledge/ingestors/repoIngestor.js
import { BaseIngestor } from '../baseIngestor.js';

// Target design: shallow `git clone`, walk tree respecting .gitignore,
// dispatch each file to CodeIngestor (or textIngestor for README/docs).
export class RepoIngestor extends BaseIngestor {
  constructor() {
    super('code');
  }

  async ingest(_source) {
    throw new Error('RepoIngestor not implemented — needs clone + file-walk + per-file CodeIngestor dispatch');
  }
}

File: server/app/knowledge/ingestors/repo_ingestor.py
from ..base_ingestor import BaseIngestor


class RepoIngestor(BaseIngestor):
    def __init__(self):
        super().__init__("code")

    async def ingest(self, source: dict) -> list[dict]:
        raise NotImplementedError("RepoIngestor not implemented — needs clone + file-walk + per-file CodeIngestor dispatch")
